### PR TITLE
[X86][MC] Keep backward compatibility in inline asm for constraints

### PR DESCRIPTION
**Pull request by KanRobert**
Not use r16-r31 with 'q','r','l' constraint for backward compatibility

### CODE REVIEW

File: llvm/test/CodeGen/X86/apx/asm-constraint.ll [hunk 1]
```diff
@@ -0,0 +1,24 @@
+; Check r16-r31 can not be used with 'q','r','l' constraint for backward compatibility.
+; RUN: not llc < %s -mtriple=x86_64-unknown-unknown -mattr=+egpr 2>&1 | FileCheck %s
+
+define dso_local void @q() {
+entry:
+; CHECK: error: inline assembly requires more registers than available
+  %0 = tail call i32 asm sideeffect "movq %rax, $0", "=q,~{rax},~{rbx},~{rcx},~{rdx},~{rdi},~{rsi},~{rbp},~{rsp},~{r8},~{r9},~{r10},~{r11},~{r12},~{r13},~{r14},~{r15},~{dirflag},~{fpsr},~{flags}"()
```
Comment (nickdesaulniers) [inline llvm/test/CodeGen/X86/apx/asm-constraint.ll:7]: you can get rid of `,~{dirflag},~{fpsr},~{flags}` here and below.

Comment (nickdesaulniers) [inline llvm/test/CodeGen/X86/apx/asm-constraint.ll:7]: if you give `%0` a name, you can drop the `entry` label, here and below.

Comment (nickdesaulniers) [reply llvm/test/CodeGen/X86/apx/asm-constraint.ll:7]: `tail` and `sideeffect` aren't necessary here.

File: llvm/test/CodeGen/X86/apx/asm-constraint.ll [hunk 2]
```diff
@@ -0,0 +1,24 @@
+; Check r16-r31 can not be used with 'q','r','l' constraint for backward compatibility.
+; RUN: not llc < %s -mtriple=x86_64-unknown-unknown -mattr=+egpr 2>&1 | FileCheck %s
+
+define dso_local void @q() {
```
Comment (nickdesaulniers) [inline llvm/test/CodeGen/X86/apx/asm-constraint.ll:4]: you can drop `dso_local` here and below.

**Review by nickdesaulniers — APPROVED** [review]

**Review by e-kud — APPROVED** [review]
LGTM. Thanks!

File: llvm/test/CodeGen/X86/apx/asm-constraint.ll (hunk 1 above)
Comment (KanRobert) [reply llvm/test/CodeGen/X86/apx/asm-constraint.ll]: Done.

**Comment by KanRobert** [comment]
Thanks for the review @nickdesaulniers @e-kud !

**Pull request merged**